### ISSUE TITLE
Restore from snapshot fails if PVC is deleted

### ISSUE THREAD

**Issue by disrani-px** [kubernetes-incubator/external-storage#705]
After deleting the PVC, restoring from its snapshot fails as it can't find the PVC anymore.

Steps to reproduce:
1. Create PVC
2. Take a snapshot of PVC
3. Delete PVC created in (1)
4. Try to create a clone PVC from snapshot in (2)
5. Create of the PVC fails since the PVC in (1) has been deleted

Logs:
```
[stork-6bd64f8969-9pk7k] I0403 00:11:49.602703       1 leaderelection.go:178] successfully acquired lease to provision for pvc default/mysql-snap-clone 
[stork-6bd64f8969-9pk7k] I0403 00:11:49.602809       1 controller.go:1080] scheduleOperation[provision-default/mysql-snap-clone[9aa99d07-36d3-11e8-98cc-0214683e8447]] 
[stork-6bd64f8969-9pk7k] time="2018-04-03T00:11:49Z" level=info msg="restore from VolumeSnapshotData k8s-volume-snapshot-75178d27-36d3-11e8-b1a0-e24f4dc55876" 
[stork-6bd64f8969-9pk7k] E0403 00:11:49.617504       1 controller.go:808] Failed to provision volume for claim "default/mysql-snap-clone" with StorageClass "stork-snapshot-sc": failed to create a PV from snapshot mysql-snapshot: Failed to retrieve PV pvc-638d7bf1-36d3-11e8-98cc-0214683e8447 from the API server: "persistentvolumes "pvc-638d7bf1-36d3-11e8-98cc-0214683e8447" not found" 
[stork-6bd64f8969-9pk7k] E0403 00:11:49.617805       1 goroutinemap.go:165] Operation for "provision-default/mysql-snap-clone[9aa99d07-36d3-11e8-98cc-0214683e8447]" failed. No retries permitted until 2018-04-03 00:11:50.117768412 +0000 UTC m=+358332.689745478 (durationBeforeRetry 500ms). Error: failed to create a PV from snapshot mysql-snapshot: Failed to retrieve PV pvc-638d7bf1-36d3-11e8-98cc-0214683e8447 from the API server: "persistentvolumes "pvc-638d7bf1-36d3-11e8-98cc-0214683e8447" not found" 
[stork-6bd64f8969-9pk7k] I0403 00:11:51.614633       1 leaderelection.go:198] stopped trying to renew lease to provision for pvc default/mysql-snap-clone, task failed 
```

/area snapshot

**Comment by discordianfish** [comment]
Just ran into the same issue. This kinda defeats the purpose of using this for backup and recovery purposes.

@wongma7 @jsafrane This seems to be a design issue I guess, correct? The provisioner needs to know things like the actual storage class that was used. I guess this should be copied to the snapshot or snapshotdata instead? Are there plans or general refactoring planned or is this something I could look into?

**Comment by disrani-px** [comment]
I actually have a simple fix for this. Will submit a PR in the next few days.

**Comment by disrani-px** [comment]
Fixed with #803
